### PR TITLE
Initial setup for AWS SDK

### PR DESCRIPTION
**Pull request by IgnacioPro**
This PR introduces the AWS SDK and reads from the bucket listed under the configuration file for syncron, 
which now will be under config folder at the root of repository.
It's only the initial setup, checking if all parameters are correct for accessing the provided bucket.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -29,8 +31,8 @@ require (
 	github.com/inconshreveable/mousetrap v1.0.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
-	github.com/stretchr/objx v0.4.0 // indirect
 	github.com/spf13/viper v1.13.0
```
Comment (jsanemet) [inline go.mod]: Run 'go mod tidy' to move 'viper' to the direct dependencies section.

File: cmd/adhoc/cmd/root.go [hunk 2]
```diff
@@ -36,17 +42,40 @@ func onPersistentPreRun(cmd *cobra.Command, args []string) error {
 		},
 		func() error {
 			// Setting up file formatting
-			viper.SetConfigFile("syncron.yaml")
+			viper.SetConfigFile("config/syncron.yaml")
```
Comment (jsanemet) [inline cmd/adhoc/cmd/root.go:45]: You will probably want to update '.gitignore' too.

File: cmd/adhoc/cmd/root.go [hunk 3]
```diff
@@ -36,17 +42,40 @@ func onPersistentPreRun(cmd *cobra.Command, args []string) error {
...
+			// Initialize a session with AWS SDK
+			// It will read from the file located at ~/.aws/credentials and syncron/config/syncron.yaml
+			logrus.Info("Starting setup for AWS s3")
+			sess, err := session.NewSession(&aws.Config{
+				Region:   aws.String(viper.GetString("s3.region")),
+				Endpoint: aws.String(viper.GetString("s3.endpoint")),
+			},
+			)
+			if err != nil {
+				fmt.Println("Error setting up AWS...", "Check config params", err)
```
Comment (jsanemet) [inline cmd/adhoc/cmd/root.go]: If the session has failed to be established, you will not want to continue with the setup. Return the error to stop flow here.

File: cmd/adhoc/cmd/root.go [hunk 4]
```diff
@@ -36,17 +42,40 @@ func onPersistentPreRun(cmd *cobra.Command, args []string) error {
...
+			} else {
+				logrus.Info("AWS setup succesful!")
+			}
+			logrus.Info("Accessing bucket...")
+			svc := s3.New(sess)
+			resp, err := svc.ListObjectsV2(&s3.ListObjectsV2Input{Bucket: aws.String(viper.GetString("bucket"))})
+			if err != nil {
+				fmt.Println("Unable to list items in bucket", err, resp)
```
Comment (jsanemet) [inline cmd/adhoc/cmd/root.go]: Same as above.

**Review by jsanemet — CHANGES_REQUESTED** [review]
Looks good for the most part. Once you start doing more with S3, you will mostly want to move this code to a module under 'pkg'.

File: cmd/adhoc/cmd/root.go (hunk 2 above)
Comment (IgnacioPro) [reply cmd/adhoc/cmd/root.go:45]: No need since .gitignore will ignore the file regardless of path.

**Review by adrianfusco — APPROVED** [review]
+1

**Review by jsanemet — APPROVED** [review]
+1

**Pull request merged**